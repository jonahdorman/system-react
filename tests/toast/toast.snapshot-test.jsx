/* eslint-env jest */
import { testDOMandHTML } from '../snapshot-helpers';

import Info from '../../examples/toast/info';
import Warning from '../../examples/toast/warning';
import ErrorAlert from '../../examples/toast/error';
import ErrorWithDetailsAlert from '../../examples/toast/error-with-details';
import Success from '../../examples/toast/success';
import CustomClassNames from '../../examples/toast/custom-class-name';

testDOMandHTML({
	name: 'Alert Info',
	test,
	Component: Info
});

testDOMandHTML({
	name: 'Alert Warning',
	test,
	Component: Warning
});

testDOMandHTML({
	name: 'Alert Error',
	test,
	Component: ErrorAlert
});

testDOMandHTML({
	name: 'Alert Error With details',
	test,
	Component: ErrorWithDetailsAlert
});

testDOMandHTML({
	name: 'Alert Success',
	test,
	Component: Success
});

testDOMandHTML({
	name: 'Alert Custom Class Name',
	test,
	Component: CustomClassNames
});
