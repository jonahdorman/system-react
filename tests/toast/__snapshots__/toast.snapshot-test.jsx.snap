// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Custom Class Name DOM Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSToastContainer
    className="this-is-the-container"
  >
    <SLDSToast
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      className="this-is-the-alert"
      labels={
        Object {
          "heading": "26 potential duplicate leads were found.",
          "headingLink": "Select Leads to Merge",
        }
      }
      variant="info"
    />
  </SLDSToastContainer>
</SLDSIconSettings>
`;

exports[`Alert Custom Class Name HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container this-is-the-container\\">
  <div class=\\"slds-notify slds-notify_toast slds-theme_info this-is-the-alert\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">info</span><span class=\\"slds-m-right_small slds-no-flex slds-align-top\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#info\\"></use></svg></span>
    <div
      class=\\"slds-notify__content\\">
      <h2 class=\\"slds-text-heading_small\\">26 potential duplicate leads were found. <a href=\\"javascript:void(0);\\">Select Leads to Merge</a></h2>
  </div><button class=\\"slds-button slds-button--icon-inverse slds-notify__close\\" title=\\"Close\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--large\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Close</span></button></div>
</div>"
`;

exports[`Alert Error DOM Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSToastContainer>
    <SLDSToast
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      labels={
        Object {
          "heading": "Can’t save lead “Sally Wong” because another lead has the same name.",
        }
      }
      variant="error"
    />
  </SLDSToastContainer>
</SLDSIconSettings>
`;

exports[`Alert Error HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_toast slds-theme_error\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">error</span><span class=\\"slds-m-right_small slds-no-flex slds-align-top\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#error\\"></use></svg></span>
    <div
      class=\\"slds-notify__content\\">
      <h2 class=\\"slds-text-heading_small\\">Can’t save lead “Sally Wong” because another lead has the same name. </h2>
  </div><button class=\\"slds-button slds-button--icon-inverse slds-notify__close\\" title=\\"Close\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--large\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Close</span></button></div>
</div>"
`;

exports[`Alert Error With details DOM Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSToastContainer>
    <SLDSToast
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      labels={
        Object {
          "details": "Here's some detail of what happened, being very descriptive and transparent.",
          "heading": "You've encountered some errors when trying to save edits to Samuel Smith.",
        }
      }
      variant="error"
    />
  </SLDSToastContainer>
</SLDSIconSettings>
`;

exports[`Alert Error With details HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_toast slds-theme_error\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">error</span><span class=\\"slds-m-right_small slds-no-flex slds-align-top\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#error\\"></use></svg></span>
    <div
      class=\\"slds-notify__content\\">
      <h2 class=\\"slds-text-heading_small\\">You&#x27;ve encountered some errors when trying to save edits to Samuel Smith. </h2>
      <p>Here&#x27;s some detail of what happened, being very descriptive and transparent.</p>
  </div><button class=\\"slds-button slds-button--icon-inverse slds-notify__close\\" title=\\"Close\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--large\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Close</span></button></div>
</div>"
`;

exports[`Alert Info DOM Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSToastContainer>
    <SLDSToast
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      labels={
        Object {
          "heading": "26 potential duplicate leads were found.",
          "headingLink": "Select Leads to Merge",
        }
      }
      onClickHeadingLink={[Function]}
      variant="info"
    />
  </SLDSToastContainer>
</SLDSIconSettings>
`;

exports[`Alert Info HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_toast slds-theme_info\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">info</span><span class=\\"slds-m-right_small slds-no-flex slds-align-top\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#info\\"></use></svg></span>
    <div
      class=\\"slds-notify__content\\">
      <h2 class=\\"slds-text-heading_small\\">26 potential duplicate leads were found. <a href=\\"javascript:void(0);\\">Select Leads to Merge</a></h2>
  </div><button class=\\"slds-button slds-button--icon-inverse slds-notify__close\\" title=\\"Close\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--large\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Close</span></button></div>
</div>"
`;

exports[`Alert Success DOM Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSToastContainer>
    <SLDSToast
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      labels={
        Object {
          "heading": Array [
            "Account ",
            <a
              href="javascript:void(0);"
            >
              ACME - 100
            </a>,
            " widgets was created.",
          ],
        }
      }
      variant="success"
    />
  </SLDSToastContainer>
</SLDSIconSettings>
`;

exports[`Alert Success HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_toast slds-theme_success\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">success</span><span class=\\"slds-m-right_small slds-no-flex slds-align-top\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#success\\"></use></svg></span>
    <div
      class=\\"slds-notify__content\\">
      <h2 class=\\"slds-text-heading_small\\">Account <a href=\\"javascript:void(0);\\">ACME - 100</a> widgets was created. </h2>
  </div><button class=\\"slds-button slds-button--icon-inverse slds-notify__close\\" title=\\"Close\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--large\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Close</span></button></div>
</div>"
`;

exports[`Alert Warning DOM Snapshot 1`] = `
<SLDSIconSettings
  iconPath="/assets/icons"
>
  <SLDSToastContainer>
    <SLDSToast
      assistiveText={
        Object {
          "closeButton": "Close",
        }
      }
      labels={
        Object {
          "heading": "Can’t share file “report-q3.pdf” with the selected users.",
        }
      }
      variant="warning"
    />
  </SLDSToastContainer>
</SLDSIconSettings>
`;

exports[`Alert Warning HTML Snapshot 1`] = `
"<div class=\\"slds-notify-container\\">
  <div class=\\"slds-notify slds-notify_toast slds-theme_warning\\" role=\\"alert\\"><span class=\\"slds-assistive-text\\">warning</span><span class=\\"slds-m-right_small slds-no-flex slds-align-top\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#warning\\"></use></svg></span>
    <div
      class=\\"slds-notify__content\\">
      <h2 class=\\"slds-text-heading_small\\">Can’t share file “report-q3.pdf” with the selected users. </h2>
  </div><button class=\\"slds-button slds-button--icon-inverse slds-notify__close\\" title=\\"Close\\" type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--large\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#close\\"></use></svg><span class=\\"slds-assistive-text\\">Close</span></button></div>
</div>"
`;
